x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/cohere/CohereEmbeddingsResponseEntity.java

```diff
@@ -0,0 +1,230 @@
+                    jsonParser,
+                    parser -> CohereEmbeddingsResponseEntity.parseEmbeddingsArray(
+                        parser,
+                        CohereEmbeddingsResponseEntity::parseEmbeddingFloatEntry
+                    )
+                );
+
+                return new TextEmbeddingResults(embeddingList);
+            } else {
+                throwUnknownToken(token, jsonParser);
+            }
+
+            // This should never be reached. The above code should either return successfully or hit the throwUnknownToken
+            // or throw a parsing exception
+            throw new IllegalStateException("Reached an invalid state while parsing the Cohere response");
+        }
+    }
+
+    private static TextEmbeddingResults parseEmbeddingsObject(XContentParser parser) throws IOException {
+        XContentParser.Token token;
+
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                var embeddingValueParser = EMBEDDING_PARSERS.get(parser.currentName());
+                if (embeddingValueParser == null) {
+                    continue;
+                }
+
+                parser.nextToken();
+                var embeddingList = XContentParserUtils.parseList(
+                    parser,
+                    listParser -> CohereEmbeddingsResponseEntity.parseEmbeddingsArray(listParser, embeddingValueParser)
+                );
+
+                return new TextEmbeddingResults(embeddingList);
+            }
+        }
+
+        throw new IllegalStateException(
+            Strings.format(
+                "Failed to find a supported embedding type in the Cohere embeddings response. Supported types are [%s]",
+                VALID_EMBEDDING_TYPES_STRING
+            )
+        );
+    }
+
+    private static TextEmbeddingResults.Embedding parseEmbeddingsArray(
+        XContentParser parser,
+        CheckedFunction<XContentParser, EmbeddingValue, IOException> parseEntry
+    ) throws IOException {
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
+        List<EmbeddingValue> embeddingValues = XContentParserUtils.parseList(parser, parseEntry);
+
+        return new TextEmbeddingResults.Embedding(embeddingValues);
+    }
+
+    private static FloatValue parseEmbeddingFloatEntry(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.currentToken();
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token, parser);
+        return new FloatValue(parser.floatValue());
+    }
+
+    private static ByteValue parseEmbeddingInt8Entry(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.currentToken();
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token, parser);
+        var parsedByte = parser.shortValue();
```

Comment:
I didn't see a `byteValue()` method. Should I add one to the parsing library? That'd probably be a follow up PR.